### PR TITLE
Nightly parallel pack installation

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25726

## Description
nightly pack installation works with multithreading.
**limitation**: can't tell if the action succeeded (exception doesn't raise from the thread).

### CODE REVIEW

File: Tests/configure_and_test_integration_instances.py [hunk 1]
```diff
@@ -800,17 +801,33 @@ def main():
...
-                client = demisto_client.configure(base_url=server_url, username=username, password=password,
-                                                  verify_ssl=False)
-                search_and_install_packs_and_their_dependencies(pack_ids, client, prints_manager, options.is_nightly)
-            except Exception as exc:
-                prints_manager.add_print_job(str(exc), print_error, 0)
-                prints_manager.execute_thread_prints(0)
-                installed_content_packs_successfully = False
+        if options.nightly:
+            threads_list = []
+            threads_prints_manager = ParallelPrintsManager(len(servers))
```
Comment (ShahafBenYakir) [inline Tests/configure_and_test_integration_instances.py:806]:
```suggestion
            threads_print_manager = ParallelPrintsManager(len(servers))
```

File: Tests/configure_and_test_integration_instances.py [hunk 2]
```diff
@@ -800,17 +801,33 @@ def main():
...
+            # For each server url we install content
+            for thread_index, server_url in enumerate(servers):
+                client = demisto_client.configure(base_url=server_url, username=username,
+                                                  password=password, verify_ssl=False)
+                t = Thread(target=install_all_content_packs,
+                           kwargs={'client': client, 'server': server_url,
+                                   'prints_manager': threads_prints_manager,
+                                   'thread_index': thread_index})
```
Comment (ShahafBenYakir) [inline Tests/configure_and_test_integration_instances.py:814]:
```suggestion
                t = Thread(target=install_all_content_packs,
                           kwargs={'client': client, 'host': server_url,
                                   'prints_manager': threads_prints_manager,
                                   'thread_index': thread_index})
```

File: Tests/Marketplace/search_and_install_packs.py [hunk 3]
```diff
@@ -308,8 +313,8 @@ def search_and_install_packs_and_their_dependencies(pack_ids, client, prints_man
         return []
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/search_and_install_packs.py:313]: Can remove this `if is_nightly` it is now redundant

**Comment by yaakovi** [comment]
triggered a nightly build: https://app.circleci.com/pipelines/github/demisto/content/17051/workflows/89a8a916-dc2b-47c7-aac3-30442387e689/jobs/59496/steps
pack installation on all demisto instances took less than 10 minutes (last night it took around 1 hour

**Comment by yaakovi** [comment]
the latest build is a nightly triggered.
latest regular build passed: https://app.circleci.com/pipelines/github/demisto/content/17162/workflows/d986413e-121c-4967-a4fe-ebcb0dd8c427/jobs/59634

force pushing

**Pull request merged**